### ISSUE TITLE
Problem with SessionState

### ISSUE THREAD

**Issue by marcoslucianops** [whitphx/streamlit-webrtc#208]
When I use SessionState in my streamlit, I getting this error

![image](https://user-images.githubusercontent.com/47423225/118550724-6889ab80-b733-11eb-9006-23c88cabb662.png)

It can be solved by updating streamlit-webrtc SessionState.py file to

```
from streamlit.report_thread import get_report_ctx
import streamlit as st


class SessionState(object):
    def __init__(self, **kwargs):
        """A new SessionState object.

        Parameters
        ----------
        **kwargs : any
            Default values for the session state.

        Example
        -------
        >>> session_state = SessionState(user_name='', favorite_color='black')
        >>> session_state.user_name = 'Mary'
        ''
        >>> session_state.favorite_color
        'black'

        """
        for key, val in kwargs.items():
            setattr(self, key, val)


@st.cache(allow_output_mutation=True)
def get_session(id, **kwargs):
    return SessionState(**kwargs)


def get(**kwargs):
    """Gets a SessionState object for the current session.

    Creates a new object if necessary.

    Parameters
    ----------
    **kwargs : any
        Default values you want to add to the session state, if we're creating a
        new one.

    Example
    -------
    >>> session_state = get(user_name='', favorite_color='black')
    >>> session_state.user_name
    ''
    >>> session_state.user_name = 'Mary'
    >>> session_state.favorite_color
    'black'

    Since you set user_name above, next time your script runs this will be the
    result:
    >>> session_state = get(user_name='', favorite_color='black')
    >>> session_state.user_name
    'Mary'

    """
    ctx = get_report_ctx()
    id = ctx.session_id
    return get_session(id, **kwargs)
```

Credits: https://gist.github.com/tvst/036da038ab3e999a64497f42de966a92#gistcomment-3484515

**Comment by whitphx** [comment]
Thank you very much!

**Comment by whitphx** [comment]
@marcoslucianops Can you share the way to reproduce this error? For example, Python version, Streamlit version, other libraries you are using with `streamlit-webrtc`, especially which also use `SessionState` internally.

**Comment by marcoslucianops** [comment]
streamlit: 0.82.0
streamlit-webrtc: 0.20.0
python: 3.6.9

SessionState: https://gist.github.com/okld/0aba4869ba6fdc8d49132e6974e2e662 (edited for my usage)

When I use state in my web.py file with app_streaming() function, I get that error in ```player_factory=create_player``` function.

It was solved using SessionState code (https://github.com/whitphx/streamlit-webrtc/issues/208#issue-893682576) in streamlit-webrtc SessionState.py file.

**Comment by whitphx** [comment]
Thank you!
So, you use your SessionState in your app, independent of the SessionState used in `streamlit-webrtc`?

**Comment by marcoslucianops** [comment]
Yes

**Comment by whitphx** [comment]
I see, it seems these two SessionState-s conflict and I'm going to fix it, not using `st.cache` because caching mechanism have some drawbacks as stated in https://gist.github.com/tvst/036da038ab3e999a64497f42de966a92#gistcomment-3484515 .

**Comment by whitphx** [comment]
This issue should be resolved in v0.20.1 along with #205
